flink-connector-mysql-cdc/src/test/java/com/ververica/cdc/connectors/mysql/source/MySqlSourceITCase.java

```diff
@@ -540,8 +614,9 @@ private void testMySqlParallelSource(
         for (int i = 0; i < captureCustomerTables.length; i++) {
             expectedBinlogData.addAll(Arrays.asList(binlogForSingleTable));
         }
+
         assertEqualsInAnyOrder(expectedBinlogData, fetchRows(iterator, expectedBinlogData.size()));
-        tableResult.getJobClient().get().cancel().get();
+        assertTrue(!hasNextData(iterator));
```

Comment:
Can use `assertFalse` to simplify assert.

Reply:
@fuyun2024 I think it's more useful to add more data check logic in methed `fetchRows`. What do you think so?

Reply:
HI @GOODBOY008  I tried to change it, but the test fails. Because putting it in the `fetchRows` method consumes binlog data ahead of time. The data actually collected by binlog is missing.